lib/proxy_api/external_ipam.rb

```diff
@@ -0,0 +1,127 @@
+module ProxyAPI
+  class ExternalIpam < ProxyAPI::Resource
+    def initialize(args)
+      @url = args[:url] + "/phpipam"
+      super args
+    end
+
+    # Retrieves the next available IP address for the specified subnet
+    # Returns: Hash with next_ip on success, hash with error otherwise
+    #   Example(success): {"cidr":"100.10.10.0/24","next_ip":"100.10.10.23"}
+    #   Example(error):   {"error":"The specified subnet does not exist in phpIPAM."}
+    def next_ip(subnet)
+      cidr = subnet.network + '/' + subnet.cidr.to_s if subnet.network.present?
+      get("/next_ip?cidr=#{cidr}")
+    rescue => e
+      raise ProxyException.new(url, e, N_("Unable to retrieve the next available IP for subnet %s from phpIPAM."), cidr)
+    end
+
+    # Adds the specified IP address to the specified subnet in phpIPAM
+    # Returns: Hash with message on success, hash with error otherwise
+    #   Example(success): {"message":"IP 100.10.10.123 added to subnet 100.10.10.0/24 successfully."}
+    #   Example(error):   {"error":"The specified subnet does not exist in phpIPAM."}
+    def add_ip_to_subnet(ip, subnet)
+      cidr = subnet.network + '/' + subnet.cidr.to_s if subnet.network.present?
+      parse(post({:ip => ip, :cidr => cidr}, "/add_ip_to_subnet"))
+    rescue => e
+      raise ProxyException.new(url, e, N_("Unable to add IP %s to the subnet %s in phpIPAM."), ip, cidr)
+    end
+
+    # Get a list of phpIPAM sections
+    # Returns: Array of sections(as json) on success, hash with error otherwise
+    #   Example(success): [{"id":"5","name":"Awesome Section","description":"A totally awesome Section",
+    #                       "masterSection":"0","permissions":"[]","strictMode":"1","subnetOrdering":"default",
+    #                       "order":null,"editDate":"2019-04-19 21:49:55","showVLAN":"1","showVRF":"1",
+    #                       "showSupernetOnly":"1","DNS":null}]]
+    #   Example(error):   {"error":"Unable to connect to phpIPAM server"}
+    def get_sections
+      get("/sections")
+    rescue => e
+      raise ProxyException.new(url, e, N_("Unable to obtain sections from phpIPAM."))
+    end
```

Comment:
Document the return value, what are permissions? Strict mode? I suggest you only return list of names, this must be generic so other implementations can do similar, they might not have sections but something else. If name is not unique in the database, then return just id + name.

Also rename phpIPAM to IPAM this must be generic.